File: src/models/payment_method.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug)]
pub enum PaymentMethod {
  Card,
  Cash,
  Izly,
  Moneo,
}

impl PaymentMethod {
  pub fn from_api(str: &str) -> PaymentMethod {
    match str {
      "Carte bancaire" => PaymentMethod::Card,
      "Espèce" => PaymentMethod::Cash,
      "IZLY" => PaymentMethod::Izly,
      "Monéo" => PaymentMethod::Moneo,
      _ => unreachable!("Unknown payment method: {}", str),
    }
  }
}

File: src/models/PaymentMethod.ts
export enum PaymentMethod {
  Card,
  Cash,
  Izly,
  Moneo
}

export const translatePaymentMethodFromAPI = (api: string): PaymentMethod => {
  switch (api) {
    case "Carte bancaire":
      return PaymentMethod.Card;
    case "Espèce":
      return PaymentMethod.Cash;
    case "IZLY":
      return PaymentMethod.Izly;
    case "Monéo":
      return PaymentMethod.Moneo;
    default:
      console.log(api);
      throw new Error(`Unknown payment method: ${api}`);
  }
};
